src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -384,7 +384,7 @@ trait ScalaSettings extends AbsScalaSettings
   }
   override def YstatisticsEnabled = Ystatistics.value.nonEmpty
 
-  val YhotStatistics = BooleanSetting("-Yhot-statistics", "Print hot compiler statistics for all relevant phases")
+  val YhotStatistics = BooleanSetting("-Yhot-statistics-enabled", s"Enable `${Ystatistics.name}` to print hot statistics.")
```

Comment:
You could make `-Ystatistics` a `ChoiceSetting` (`-Ystatistics:{none|cold|all}`), but that's not a blocker for merging this PR.